internal/pkg/telemetry/ddprovider.go

```diff
@@ -11,40 +13,41 @@ import (
 	"github.com/keboola/keboola-as-code/internal/pkg/service/common/servicectx"
 )
 
-type singleTracerProvider struct {
-	tracer trace.Tracer
+type wrappedDDTracerProvider struct {
+	*ddotel.TracerProvider
+	lock    *sync.Mutex
+	tracers map[string]trace.Tracer
 }
```

Comment:
Odstranenie workaround-u.

Nemusime uz riesit aby sme vratili vzdy rovnaku instanciu tracer, to je uz v DD fixnute.

wrapper ale stale potrebujeme, kedze chceme pouzit upravenu verziu `span.RecordError`, vid `ddspan.go` subor.


`wrappedDDTracerProvider -> wrappedDDTracer -> wrappedDDSpan`:
- teda `wrappedDDTracerProvider` vrati `wrappedDDTracer` cachovany podla `name` v map
- `wrappedDDTracer.Start` vrati `wrappedDDSpan`, ktory upravuje metodu `RecordError`.